docs/madr/decisions/015-retry-policy.md

```diff
@@ -0,0 +1,223 @@
+# Retry policy
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4738
+
+## Context and Problem Statement
+
+We want to create a new policy for retries that uses TargetRef [matching](https://github.com/kumahq/kuma/blob/22c157d4adac7f518b1b49939c7e9ea4d2a1876c/docs/madr/decisions/005-policy-matching.md).
+
+### Specification 
+
+#### Name
+
+MeshRetry
+
+#### From level
+
+Even though it's possible to place a retry policy on the inbound routes, 
+it doesn't seem like there is a value in it.
+Retries create an additional load on the service,
+that's why it's not in the interest of service owners to place retries on the inbounds.
+Most of the time retrying of the request is a client's concern.
+
+#### Top level
+
+Top-level targetRef can have the following kinds:
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute
```

Comment:
```suggestion
  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
```

Reply:
How do we configure retries for MeshGateway?

Reply:
`MeshService`. Or `MeshGateway` which we should support anyway. In general we don't support `MeshGatewayRoute` atm, IMO it's not necessary to include it here until we add it in a comprehensive way